ledger/sandbox-classic/src/main/scala/platform/sandbox/ConfigConverter.scala

```diff
@@ -0,0 +1,106 @@
+      conflictCheckingEnabled = true,
+      implicitPartyAllocation = sandboxConfig.implicitPartyAllocation,
+      authService = sandboxConfig.authService,
+      timeProviderType =
+        sandboxConfig.timeProviderType.getOrElse(SandboxConfig.DefaultTimeProviderType),
+      // TODO SoX-to-sandbox-classic: Dedicated submissionBufferSize CLI param for sanbox-classic
+      submissionBufferSize = sandboxConfig.maxParallelSubmissions,
+      // TODO SoX-to-sandbox-classic: Dedicated submissionBufferSize CLI param for sanbox-classic
+      maxDedupSeconds = BridgeConfigProvider.defaultExtraConfig.maxDedupSeconds,
+    )
+
+    val allowedLanguageVersions = sandboxConfig.engineMode match {
+      case EngineMode.Stable => LanguageVersion.StableVersions
+      case EngineMode.EarlyAccess => LanguageVersion.EarlyAccessVersions
+      case EngineMode.Dev => LanguageVersion.DevVersions
+    }
+
+    Config[BridgeConfig](
+      mode = Mode.Run,
+      ledgerId = sandboxConfig.ledgerIdMode match {
+        case LedgerIdMode.Static(ledgerId) => ledgerId.unwrap
+        case LedgerIdMode.Dynamic => maybeLedgerId.getOrElse(UUID.randomUUID().toString)
+      },
+      commandConfig = sandboxConfig.commandConfig,
+      submissionConfig = sandboxConfig.submissionConfig,
+      tlsConfig = sandboxConfig.tlsConfig,
+      participants = Seq(
+        singleCombinedParticipant
+      ),
+      maxInboundMessageSize = sandboxConfig.maxInboundMessageSize,
+      configurationLoadTimeout = sandboxConfig.configurationLoadTimeout,
+      eventsPageSize = sandboxConfig.eventsPageSize,
+      eventsProcessingParallelism = sandboxConfig.eventsProcessingParallelism,
+      acsIdPageSize = sandboxConfig.acsIdPageSize,
+      acsIdFetchingParallelism = sandboxConfig.acsIdFetchingParallelism,
+      acsContractFetchingParallelism = sandboxConfig.acsContractFetchingParallelism,
+      acsGlobalParallelism = sandboxConfig.acsGlobalParallelism,
+      acsIdQueueLimit = sandboxConfig.acsIdQueueLimit,
+      stateValueCache = caching.WeightedCache.Configuration.none,
+      lfValueTranslationEventCache = sandboxConfig.lfValueTranslationEventCacheConfiguration,
+      lfValueTranslationContractCache = sandboxConfig.lfValueTranslationContractCacheConfiguration,
+      seeding = sandboxConfig.seeding,
+      metricsReporter = sandboxConfig.metricsReporter,
+      metricsReportingInterval = sandboxConfig.metricsReportingInterval.toJava,
+      allowedLanguageVersions = allowedLanguageVersions,
+      // Enabled by default.
+      enableMutableContractStateCache = true,
+      // TODO SoX-to-sandbox-classic: Add configurable flag for sandbox-classic
+      enableInMemoryFanOutForLedgerApi = false,
+      maxDeduplicationDuration = sandboxConfig.maxDeduplicationDuration,
+      extra = extraBridgeConfig,
+      enableSelfServiceErrorCodes = sandboxConfig.enableSelfServiceErrorCodes,
+      userManagementConfig = sandboxConfig.userManagementConfig,
+    )
```

Comment:
`--scenario` as well as `--eager-package-loading`  command-line parameter is ignored, should we remove it from the `SandboxConfig` case class?

`--database-connection-pool-size` mapped to `databaseConnectionPoolSize`is a parameter that exists on sandboxnext, daml-on-sql but not on sandbox-classic. I suggest adding it to the command line so that we have a replacement for the `daml-on-sql` as well

`--profile-dir` and `--stack-traces` parameters available in sandbox-classic are dropped in current implementation. Could you wire them up?

`--sql-start-mode` parameter mapped to `sqlStartMode` exists on daml-on-sql and can be mapped to IndexerStartupMode. Could you add it too?

Reply:
> --scenario as well as --eager-package-loading command-line parameter is ignored, should we remove it from the SandboxConfig case class?

I didn't consider any removals from `SandboxConfig` as part of this PR since this config is also used by `sandbox-next`. 

> --database-connection-pool-size mapped to databaseConnectionPoolSizeis a parameter that exists on sandboxnext, daml-on-sql but not on sandbox-classic. I suggest adding it to the command line so that we have a replacement for the daml-on-sql as well

Wired `database-connection-pool-size` to `api-server-connection-pool-size`

> --profile-dir and --stack-traces parameters available in sandbox-classic are dropped in current implementation. Could you wire them up?

because the `kvutils.app` `Config` does not have them. I'll wire them

Reply:
> --sql-start-mode parameter mapped to sqlStartMode exists on daml-on-sql and can be mapped to IndexerStartupMode. Could you add it too?

This one is default `migrateAndStart` in the kvutils `Runner` and `SandboxOnXRunner` and not available as option in the `Config`. I'd defer this change in a follow-up PR so we keep the changes to this one to a decent size.

Reply:
I recommend decoupling SoX from kvutils `Config` soon; we want to move kvutils out of this repository so it's going to disappear eventually anyway.